src/main/java/core/basesyntax/strategy/OperationStrategyImpl.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.service.FruitTransaction;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private Map<FruitTransaction.Operation, OperationHandler> handlerMap;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation, OperationHandler> handlerMap) {
+        this.handlerMap = handlerMap;
+    }
+
+    @Override
+    public OperationHandler get(FruitTransaction.Operation operation) {
+        return handlerMap.get(operation);
```

Comment:
The `get` method may return null if the operation is not found in the `handlerMap`. Consider adding a check to handle this case, possibly by throwing an exception or returning a default handler, to avoid potential `NullPointerException` in the calling code.